src/ot/layout/common.rs

```diff
@@ -0,0 +1,835 @@
+//! Common tables for OpenType layout.
+
+use std::cmp::Ordering;
+use std::convert::TryFrom;
+
+use ttf_parser::parser::{
+    FromData, LazyArray16, LazyArray32, Offset, Offset16, Offset32, Offsets16, Stream, TryNumFrom,
+};
+use ttf_parser::GlyphId;
+
+use crate::font::Font;
+use crate::Tag;
+
+/// A GSUB or GPOS table.
+#[derive(Clone, Copy, Debug)]
+pub struct SubstPosTable<'a> {
+    scripts: RecordList<'a>,
+    features: RecordList<'a>,
+    lookups: LookupList<'a>,
+    variations: Option<FeatureVariations<'a>>,
+}
+
+#[allow(dead_code)]
+impl<'a> SubstPosTable<'a> {
+    pub fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+
+        let major_version = s.read::<u16>()?;
+        let minor_version = s.read::<u16>()?;
+        if major_version != 1 {
+            return None;
+        }
+
+        let scripts = RecordList::parse(s.read_at_offset16()?)?;
+        let features = RecordList::parse(s.read_at_offset16()?)?;
+        let lookups = LookupList::parse(s.read_at_offset16()?)?;
+
+        let mut variations = None;
+        if minor_version >= 1 {
+            variations = FeatureVariations::parse(s.read_at_offset32()?);
+        }
+
+        Some(Self { scripts, features, lookups, variations })
+    }
+
+    // Scripts
```

Comment:
I don't think this comments are necessary.

Reply:
I'll remove it.